dimod/reference/composites/fixedvariable.py

```diff
@@ -0,0 +1,103 @@
+import dimod
+from dimod.roof_duality import fix_variables
+from dimod.response import SampleSet
+import numpy as np
+
+__all__ = ['FixedVariableComposite']
+
+class FixedVariableComposite(dimod.ComposedSampler):
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+
+        return self._children
+
+    @property
+    def parameters(self):
+
+        return self.child.parameters.copy()
+
+    @property
+    def properties(self):
+
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, source_bqm, fixed_variables=None,
+               **parameters):
+
+        child = self.child
+        if fixed_variables is None:
+            fixed_variables = fix_variables(source_bqm)
```

Comment:
Not sure we want to do this automatically. Maybe toggled by a `use_roof_duality` flag?

Reply:
I assume if they are calling fixedvariablecomposite without the fixed_variables they want us to decide

Reply:
Ok, fair point. Though the case could be made that we would want to run that even if they do provide us with some variables to fix.

Reply:
this way we allow both options... if someone is experimenting its a nice feature to be able to define your own fixed variables without any decision making on our end.